### ISSUE TITLE
Pub upgrade succeeds but pub get fails with version conflicts? 

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/pub#1750]
you can repro this by checking out the [angular repo](https://github.com/dart-lang/angular), and running `pub get` in the `angular_router` directory.

You should see the following error:

```
Package build has no versions that match >=0.10.0 <0.12.0 derived from:
- angular 5.0.0-alpha+1 depends on version >=0.10.0 <0.12.0
- angular_compiler 0.4.0-alpha+1 depends on version >=0.10.0 <0.12.0
```

However, if you do a `pub upgrade` it succeeds. Here is the example output for the angular_router package:

```
 analyzer 0.31.0-alpha.2
  angular 5.0.0-alpha+1
  angular_ast 0.4.0-alpha+2 (0.4.0-alpha.0 available)
  angular_compiler 0.4.0-alpha+1
  angular_test 1.0.2-alpha+1
  ansicolor 0.0.9
  archive 1.0.32
  args 0.13.7 (1.1.0 available)
  async 1.13.3 (2.0.1 available)
  barback 0.15.2+13
  bazel_worker 0.1.8
  boolean_selector 1.0.2
  build 0.11.1
  build_barback 0.4.0+2
  build_compilers 0.1.0
  build_config 0.1.1
  build_runner 0.6.1
  build_test 0.9.1
  built_collection 1.6.2 (2.0.0 available)
  built_value 4.3.3 (4.4.0 available)
  charcode 1.1.1
  cli_util 0.1.2+1
  code_builder 2.3.0
  code_transformers 0.5.1+3
  collection 1.14.3
  convert 2.0.1
  crypto 2.0.2+1
  csslib 0.14.1
  dart_style 1.0.8
  fixnum 0.10.5 (0.10.6 available)
  front_end 0.1.0-alpha.7
  glob 1.1.5
  html 0.13.2
  http 0.11.3+16
  http_multi_server 2.0.4
  http_parser 3.1.1
  intl 0.15.2
  io 0.3.0
  isolate 1.1.0
  js 0.6.1
  json_annotation 0.2.1
  kernel 0.3.0-alpha.4
  logging 0.11.3+1
  matcher 0.12.1+4
  meta 1.1.2
  mime 0.9.4
  node_preamble 1.4.0
  package_config 1.0.3
  package_resolver 1.0.2
  pageloader 2.2.6
  path 1.5.1
  plugin 0.2.0+2
  pool 1.3.3
  protobuf 0.6.0
  pub_semver 1.3.2
  quiver 0.25.0 (0.26.2 available)
  quiver_hashcode 1.0.0
  scratch_space 0.0.1+2
  shelf 0.7.1
  shelf_packages_handler 1.0.3
  shelf_static 0.2.6
  shelf_web_socket 0.2.2
  source_gen 0.7.2+1
  source_map_stack_trace 1.1.4
  source_maps 0.10.4
  source_span 1.4.0
  stack_trace 1.9.1
  stream_channel 1.6.2
  stream_transform 0.0.9
  string_scanner 1.0.2
  term_glyph 1.0.0
  test 0.12.28
  tuple 1.0.1
  typed_data 1.1.5
  unittest 0.11.7 (0.12.4+1 available)
  utf 0.9.0+3
  watcher 0.9.7+4
  web_socket_channel 1.0.6
  webdriver 1.2.3 (2.0.0-beta available)
  yaml 2.1.13
```

Even after a successful `pub upgrade`, a subsequent `pub get` will again fail.

**Comment by nex3** [comment]
This is weird and bad, but I'm closing in on #912 so I'm going to hope that fixes it.

**Comment by matanlurey** [comment]
I'm also seeing this in `angular_test`.

**Comment by 4cm4k1** [comment]
FWIW, I no longer have this issue as of `2.0.0-dev.23`. 🙂

**Comment by natebosch** [comment]
I think this was resolved by the #912
